apps/src/templates/teacherDashboard/TeacherDashboard.jsx

```diff
@@ -19,142 +19,146 @@ import StandardsReport from '../sectionProgress/standards/StandardsReport';
 import SafeMarkdown from '@cdo/apps/templates/SafeMarkdown';
 import i18n from '@cdo/locale';
 
-class TeacherDashboard extends Component {
-  static propTypes = {
-    studioUrlPrefix: PropTypes.string.isRequired,
-    sectionId: PropTypes.number.isRequired,
-    sectionName: PropTypes.string.isRequired,
-    studentCount: PropTypes.number.isRequired,
-    coursesWithProgress: PropTypes.array.isRequired,
+function TeacherDashboard({
+  studioUrlPrefix,
+  sectionId,
+  sectionName,
+  studentCount,
+  coursesWithProgress,
+  location
+}) {
+  const usePrevious = value => {
+    const ref = React.useRef();
+    React.useEffect(() => {
```

Comment:
Why do you need a `useEffect` here? I'm not familiar with this pattern either so trying to understand

Reply:
oh wait is it to make sure it's only set once?

Reply:
I'm honestly not entirely sure myself- I was borrowing a pattern I found here but could use to learn a little more about it https://stackoverflow.com/questions/45373742/detect-route-change-with-react-router

Reply:
I should say- whatever it's doing, it's setting the correct amount of firehose events with the correct previous and new tabs